js/ui/windowManager.js

```diff
@@ -985,6 +1103,111 @@ var DesktopOverlay = new Lang.Class({
+    },
+
+    vfunc_attach_effect: function(event, effect) {
+        let effects = _ALLOWED_ANIMATIONS_FOR_EVENTS[event];
+
+        if (!effects) {
+            throw new GLib.Error(AnimationsDbus.error_quark(),
+                                 AnimationsDbus.Error.UNSUPPORTED_EVENT_FOR_ANIMATION_SURFACE,
+                                 `Surface does not support event ${event}`);
+        }
+
+        if ((effects || []).indexOf(effect.name) == -1) {
+            throw new GLib.Error(AnimationsDbus.error_quark(),
+                                 AnimationsDbus.Error.UNSUPPORTED_EVENT_FOR_ANIMATION_EFFECT,
+                                 `Effect ${effect.name} can't be used on event ${event}`);
+        }
+
+        return effect.bridge.createActorPrivate(this.actor);
+    },
+
+    vfunc_detach_effect: function(event, attached_effect) {
+        attached_effect.remove();
+    },
+
+    vfunc_get_title: function() {
+        return this.actor.meta_window.title;
+    },
+
+    vfunc_get_geometry: function() {
+        return new GLib.Variant("(iiii)", [
+            this.actor.x,
+            this.actor.y,
+            this.actor.width,
+            this.actor.height
+        ]);
+    },
+
+    vfunc_get_available_effects: function() {
+        return new GLib.Variant("a{sv}", Object.keys(_ALLOWED_ANIMATIONS_FOR_EVENTS).reduce(function(acc, key) {
+            acc[key] = new GLib.Variant("as", _ALLOWED_ANIMATIONS_FOR_EVENTS[key]);
+            return acc
+        }, {}));
+    }
+});
+
+// ShellWindowManagerAnimationsFactory
+//
+// An implementation of AnimationsDbus.ServerEffectFactory
+// which implements the create_effect() method. When a
+// caller tries to create an animation, the name is looked up
+// here an a corresponding AnimationsDbus.ServerEffectBridge
+// implementation is returned if one is available for that
+// effect name, which represents the metaclass for that
+// effect as it exists on the shell side.
+const ShellWindowManagerAnimationsFactory = new Lang.Class({
+    Name: 'ShellWindowManagerAnimationsFactory',
+    Implements: [ AnimationsDbus.ServerEffectFactory ],
+    Extends: GObject.Object,
+
+    _init: function() {
+        this.parent();
+    },
+
+    vfunc_create_effect: function(name, settings) {
+        switch (name) {
+            case 'wobbly':
+                return new ControllableShellWobblyEffect();
+            case 'gsettings-wobbly':
+                return new GSettingsShellWobblyEffect();
+            default:
+                throw new GLib.Error(AnimationsDbus.error_quark(),
+                                     AnimationsDbus.Error.NO_SUCH_EFFECT,
+                                     `No such effect ${name}`);
+        }
+    }
+});
+
+function getAnimatableWindowActors() {
+    return global.get_window_actors().filter(w => ([
+        Meta.WindowType.NORMAL,
+        Meta.WindowType.DIALOG,
+        Meta.WindowType.MODAL_DIALOG
+    ].indexOf(w.meta_window.get_window_type()) !== -1));
+}
```

Comment:
Is there no other function we can already use for this?

Reply:
I didn't see one, though the shell code is pretty messy so I wouldn't be surprised if there was and I just hadn't found it yet :/

Reply:
In any event, the code in `windowManager.js` around window types is all over the place already. We can punt a cleanup to a later stage.